saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncPersister.kt

```diff
@@ -125,7 +115,8 @@ class SavedSitesSyncPersister @Inject constructor(
 
     private class Adapters {
         companion object {
-            private val moshi = Moshi.Builder().build()
+            private val moshi = Moshi.Builder()
+                .add(KotlinJsonAdapterFactory()).build()
```

Comment:
@malmstein I had to add this factory here in order moshi can validate mandatory fields in the response. Otherwise, it will never crash when parsing (that's why we had those other checks in place).